core/src/client/async_client/mod.rs

```diff
@@ -378,13 +398,202 @@ impl SubscriptionClientT for Client {
 	}
 }
 
+/// Handle backend messages.
+///
+/// Returns error if the main background loop should be terminated.
+async fn handle_backend_messages<S: TransportSenderT, R: TransportReceiverT>(
+	message: Option<Result<ReceivedMessage, R::Error>>,
+	manager: &mut RequestManager,
+	sender: &mut S,
+	max_notifs_per_subscription: usize,
+) -> Result<(), Error> {
+
+	// Handle raw messages of form `ReceivedMessage::Bytes` (Vec<u8>) or ReceivedMessage::Data` (String).
+	async fn handle_recv_message<S: TransportSenderT>(
```

Comment:
So I am under the impression the reason this is nested is this `fn` is only used locally within the scope of `handle_backend_messages`? I quite like it actually as it distinquishes the difference between handling frontend messages and backend message. 👍  Nice cleanup from it earlier being inside of the `fn background_task`. It's way easier to read/digest.